### PR TITLE
feat(Fault Management): allow fault providers to define shelve durations

### LINKED ISSUE #5814: [Build] Move off of codecov npm package
#### Summary

History: we needed to do extensive local debugging with codecov to ensure that what was run locally matched exactly what was run in circleci/githubactions, so we implemented a soon-to-be-deprecated npm package. We should remove it.

We're using codecov's npm package which is deprecated. We need to move to the officially supported pattern.

https://github.com/marketplace/actions/codecov
https://about.codecov.io/blog/codecov-uploader-deprecation-plan/

### LINKED ISSUE #7853: [Fault Management] Providers should be able to specify shelve durations
**Is your feature request related to a problem? Please describe.**
Currently, shelve durations are hard-coded and only provide a few different options.

**Describe the solution you'd like**
Fault providers should be able to define their own shelve durations. An optional API method such as `getShelveDurations` can be defined by the fault provider and return a list of options with label and duration (in ms):

```js
[
  {
    name: '1 minute',
    duration: 60 * 60 * 1000
  },
  {
    name: '5 minutes',
    duration: 5 * 60 * 60 * 1000
  }
]
```

### PR DESCRIPTION
**Pull request by ozyx**
Closes #7853
Closes #5814 
 

### Describe your changes:

- Adds a new, optional method `getShelveDurations()` to the Fault Management API that allows providers to define their own shelve durations. If no method is defined, a default list of shelve durations will be provided.
- Adds some additional docs and JSDoc types for Fault Management API
- Refactors the Fault Management e2e suite to use web-first assertions and locators
- Build: removes the `codecov` npm package dependency as it has been deprecated for some time.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this a [notable change](../docs/src/process/release.md) that will require a special callout in the release notes? For example, will this break compatibility with existing APIs or projects that consume these plugins?

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?

### CODE REVIEW

File: src/plugins/faultManagement/FaultManagementView.vue [hunk 1]
```diff
@@ -318,19 +330,16 @@ export default {
...
       } else {
         shelveData = {
           shelved: false
         };
       }
 
-      Object.values(faults).forEach((selectedFault) => {
-        this.openmct.faults.shelveFault(selectedFault, shelveData);
-      });
+      await Promise.all(
```
Comment (scottbell) [inline src/plugins/faultManagement/FaultManagementView.vue:340]: nice catch

**Review by scottbell — COMMENTED** [review]
@ozyx Can you add some testing instructions please? Code looks good.

**Comment by ozyx** [comment]
> @ozyx Can you add some testing instructions please? Code looks good.

yep! will do

**Comment by ozyx** [comment]
> @ozyx Can you add some testing instructions please? Code looks good.

@scottbell done!

File: example/faultManagement/utils.js [hunk 2]
```diff
@@ -13,7 +35,8 @@ const getRandom = {
 
       val = num;
       severity = SEVERITIES[severityIndex - 1];
-      time = num;
+      // Subtract `num` from the timestamp so that the faults are in order
+      time = 14159040000 - num; // Mon, 21 Jul 1969 02:56:00 GMT 🌔👨‍🚀👨‍🚀👨‍🚀
```
Comment (shefalijoshi) [inline example/faultManagement/utils.js:39]: Change this hard coded timestamp `14159040000` to a const.

**Review by shefalijoshi — COMMENTED** [review]
One small change. Very nice documentation!

**Review by shefalijoshi — APPROVED** [review]
Looks good. Thanks.

**Pull request merged**